lsp/src/Language/LSP/Server/Core.hs

```diff
@@ -441,12 +441,18 @@ freshLspId = do
 -- ---------------------------------------------------------------------
 
 -- | The current configuration from the client as set via the @initialize@ and
--- @workspace/didChangeConfiguration@ requests.
+-- @workspace/didChangeConfiguration@ requests, as well as by calls to
+-- 'setConfig'.
 getConfig :: MonadLsp config m => m config
 getConfig = getsState resConfig
 
 {-# INLINE getConfig #-}
 
+setConfig :: MonadLsp config m => (config -> (a,config)) -> m a
```

Comment:
This isn't really what we'd normally call `setConfig`, which would have type `config -> m ()`. I think we should either expose that function, or call this `stateConfig` or something weird like that. (the whole state manipulation dance here is a bit odd, but that's how it is...)

Reply:
Understood, I've updated the PR accordingly.